From 8d3f2c5c7d7139b4aec89f1c2cb2ce1e99aa5f53 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Thu, 5 Sep 2019 17:44:15 -0700
Subject: [PATCH 402/767] drivers: cleanup el3 handler only if parent process
 exits

commit 3c1276ffc89556880183f9fa5fb30491214adb91 from
git@git.assembla.com:cavium/WindRiver.linux.git

We have EL0 interrupt handling in various drivers. There are cleanup
handlers that are called when a process is killed. These handlers are
also being called when child process exits. So do not cleanup if its
child.

Change-Id: Ie068860174c71155283482d414184abcd04728b6
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/15390
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/18209
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/gpio/gpio-thunderx.c | 3 +--
 drivers/misc/otx_bphy_ctr.c  | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index e573e469d429..53b593642082 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -231,8 +231,7 @@ static void cleanup_el3_irqs(struct task_struct *task)
 	for (i = 0; i < MAX_GPIO; i++) {
 		if (gpio_installed[i] &&
 		    gpio_installed_tasks[i] &&
-		    ((gpio_installed_tasks[i] == task) ||
-			(gpio_installed_tasks[i] == task->group_leader))) {
+		    (gpio_installed_tasks[i] == task)) {
 			pr_alert("Exiting, removing handler for GPIO %d\n",
 				 i);
 			__remove_el3_inthandler(i);
diff --git a/drivers/misc/otx_bphy_ctr.c b/drivers/misc/otx_bphy_ctr.c
index 7818ca69d345..0bdb1616b7c8 100644
--- a/drivers/misc/otx_bphy_ctr.c
+++ b/drivers/misc/otx_bphy_ctr.c
@@ -167,8 +167,7 @@ static void cleanup_el3_irqs(struct task_struct *task)
 	for (i = 0; i < MAX_IRQ; i++) {
 		if (irq_installed[i] &&
 		    irq_installed_tasks[i] &&
-		    ((irq_installed_tasks[i] == task) ||
-			(irq_installed_tasks[i] == task->group_leader))) {
+		    (irq_installed_tasks[i] == task)) {
 			pr_alert("Exiting, removing handler for BPHY IRQ %d\n",
 				 i);
 			__remove_el3_inthandler(i);
-- 
2.31.1

